src/test/java/uk/gov/pay/directdebit/webhook/gocardless/resources/WebhookGoCardlessResourcePaymentActionsIT.java

```diff
@@ -0,0 +1,219 @@
+import uk.gov.pay.directdebit.payers.fixtures.PayerFixture;
+import uk.gov.pay.directdebit.payments.fixtures.GatewayAccountFixture;
+import uk.gov.pay.directdebit.payments.fixtures.PaymentFixture;
+import uk.gov.pay.directdebit.payments.model.GoCardlessPaymentId;
+import uk.gov.pay.directdebit.payments.model.PaymentState;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalToJson;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.stubFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.verify;
+import static com.google.common.base.Charsets.UTF_8;
+import static io.restassured.RestAssured.given;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.Response.Status.OK;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static uk.gov.pay.directdebit.notifications.model.EmailPayload.EmailTemplate;
+
+@RunWith(DropwizardJUnitRunner.class)
+@DropwizardConfig(app = DirectDebitConnectorApp.class, config = "config/test-it-config.yaml")
+public class WebhookGoCardlessResourcePaymentActionsIT {
+
+@Rule
+    public WireMockRule wireMockRuleForAdminUsersPort = new WireMockRule(10110);
+
+    private final GatewayAccountFixture gatewayAccountFixture = GatewayAccountFixture.aGatewayAccountFixture()
+            .withPaymentProvider(PaymentProvider.GOCARDLESS)
+            .withOrganisation(GoCardlessOrganisationId.valueOf("OR00003V8M32F0"));
+
+    private final MandateFixture mandateFixture = MandateFixture.aMandateFixture()
+            .withGatewayAccountFixture(gatewayAccountFixture);
+
+    private final PayerFixture payerFixture = PayerFixture.aPayerFixture()
+            .withMandateId(mandateFixture.getId());
+
+    private final PaymentFixture paymentFixture = PaymentFixture.aPaymentFixture()
+            .withMandateFixture(mandateFixture)
+            .withPaymentProviderId(GoCardlessPaymentId.valueOf("PM000JWCBM6ABD"));
+
+    @DropwizardTestContext
+    private TestContext testContext;
+
+    @Before
+    public void setUp() {
+        gatewayAccountFixture.insert(testContext.getJdbi());
+        mandateFixture.insert(testContext.getJdbi());
+        payerFixture.insert(testContext.getJdbi());
+        paymentFixture.insert(testContext.getJdbi());
+    }
+
+    @Test
+    @Ignore
+    public void customerApprovalDeniedChangesStateToCustomerApprovalDenied() throws IOException {
+        postWebhook("gocardless-webhook-payment-customer-approval-denied.json",
+                "d8040d2583728992037e0c345cc1818c264b192ec62c6a0fd2ac654ac7f4823e");
```

Comment:
As per the mandate test I think it would be beneficial to have a helper method generate the signature so we can more easily modify and add new fixtures without having to pre-calculate it and hard-code it (no doubt you could do it in a fashion that its used by both tests).